### PR TITLE
feat: added event-iterator package

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: packages/event-iterator/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+**Note:** While this section uses `require`, the imports match 1:1 with ESM imports. For example `const { EventIterator } = require('@sapphire/event-iterator')` equals `import { EventIterator } from '@sapphire/event-iterator'`.
+
+### Basic Usage
+
+```ts
+// Import the Bucket class
+const { EventIterator } = require('@sapphire/event-iterator');
+
+// Define a bucket with 1 usage every 5 seconds
```
Comment (vladfrangu) [inline packages/event-iterator/README.md:58]: Bucket this di

File: packages/event-iterator/README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+emitter.emit('message', { id: 'bar' });
+emitter.emit('message', { id: 'baz' });
+
+for await (const message of iterator) {
+	console.log(message);
+	// Logs: { id: 'foo' }
+	// Logs: { id: 'bar' }
+}
+
+// The iterator ended at second item, { id: 'baz' } is not collected.
```
Comment (vladfrangu) [inline packages/event-iterator/README.md:71]:
```suggestion
// The iterator ended at second item, { id: 'baz' } is not collected due to the specified limit.
```

File: packages/event-iterator/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+/**
+ * An EventIterator, used for asynchronously iterating over received values.
+ */
+export class EventIterator<V extends unknown[]> implements AsyncIterableIterator<V> {
+	/**
+	 * The emitter to listen to.
+	 */
+	public readonly emitter: EventEmitter;
+
+	public readonly event: string;
```
Comment (vladfrangu) [inline packages/event-iterator/src/index.ts:35]: Docstring

File: packages/event-iterator/README.md (hunk 1 above)
Comment (cfanoulis) [inline packages/event-iterator/README.md:58]: uhm... ?
```suggestion
```

File: packages/event-iterator/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+## Usage
+
+**Note:** While this section uses `require`, the imports match 1:1 with ESM imports. For example `const { EventIterator } = require('@sapphire/event-iterator')` equals `import { EventIterator } from '@sapphire/event-iterator'`.
+
+### Basic Usage
+
+```ts
+// Import the Bucket class
```
Comment (cfanoulis) [inline packages/event-iterator/README.md:55]: you got the readme's mixed up
```suggestion
```

**Review by cfanoulis — CHANGES_REQUESTED** [review]
seems like you got over-excited with leaky buckets?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by cfanoulis — DISMISSED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by cfanoulis — APPROVED** [review]

**Pull request merged**